src/rgw/rgw_rest_s3.cc

```diff
@@ -418,6 +418,56 @@ int RGWGetObj_ObjStore_S3::get_decrypt_filter(std::unique_ptr<RGWGetObj_Filter>
   }
   return res;
 }
+int RGWGetObj_ObjStore_S3::verify_requester(const rgw::auth::StrategyRegistry& auth_registry) 
+{
+  int ret = -EINVAL;
+  ret = RGWOp::verify_requester(auth_registry);
+  if(!s->user->get_caps().check_cap("amz-cache", RGW_CAP_READ) && !ret && s->info.env->exists("HTTP_X_AMZ_CACHE"))
+    ret = override_range_hdr(auth_registry);
+  return ret;
+}
+int RGWGetObj_ObjStore_S3::override_range_hdr(const rgw::auth::StrategyRegistry& auth_registry)
+{
+  int ret = -EINVAL;
+  ldpp_dout(this,2) << "cache override headers" << dendl;
+  try {
+      RGWEnv* rgw_env = const_cast<RGWEnv *>(s->info.env);
+      // Maybe we need to change env inside req_info to non const var, maybe we can use const_cast or maybe keep it that way (changing rgw_env from the ref)
+      const char *backup_range = rgw_env->get("HTTP_RANGE");
+      const char hdrs_split[2] = {(char)178,'\0'};
+      const char kv_split[2] = {(char)177,'\0'};
+      std::string cache_hdr(rgw_env->get("HTTP_X_AMZ_CACHE")); // we are calling process_cache_request method only if the header exists 
+      vector<string> cache_hdrs, cache_kvs;
+      boost::split(cache_hdrs,cache_hdr,boost::is_any_of(hdrs_split));
```

Comment:
there's a `ceph::split()` in common/split.h that can avoid allocating vectors here. for example,
```c++
  const char hdrs_split[2] = {(char)178,'\0'};
  const char kv_split[2] = {(char)177,'\0'};
  const char* cache_hdr = rgw_env->get("HTTP_X_AMZ_CACHE");

  for (std::string_view hdr : ceph::split(cache_hdr, hdrs_split)) {
    auto kv = ceph::split(hdr, kv_split);
    auto k = kv.begin();
    if (std::distance(k, kv.end()) != 2) {
      return -EINVAL;
    }
    auto v = std::next(k);

    std::string key = "HTTP_";
    key.append(*k);
    boost::replace_all(key, "-", "_");
    rgw_env->set(std::move(key), *v);
  }
```

Reply:
Used it, Thank you